prov/psm2/src/psmx2_init.c

```diff
@@ -70,9 +70,30 @@ int	 psmx2_tag_layout_locked = 0;
 
 static void psmx2_init_env(void)
 {
+	char *ompi_job_key;
+	psm2_uuid_t uuid = {};
+	unsigned long long *u = (unsigned long long *)uuid;
+
 	if (getenv("OMPI_COMM_WORLD_RANK") || getenv("PMI_RANK") || getenv("PMIX_RANK"))
 		psmx2_env.name_server = 0;
 
+	/*
+	 * OpenMPI 4.x passes the job key via the environment variable, but the
+	 * format is different. Perform format conversion and use it as the
+	 * default uuid. This will be overridden if FI_PSM2_UUID is set, and
+	 * both have lower priority than the auth_key passed via ep_attr.
+	 */
+	ompi_job_key = getenv("OMPI_MCA_orte_precondition_transports");
+	if (ompi_job_key) {
+		FI_INFO(&psmx2_prov, FI_LOG_CORE,
+			"OMPI job key: %s\n", ompi_job_key);
```

Comment:
```suggestion
			"Open MPI job key: %s\n", ompi_job_key);
```

Please use "Open MPI" instead of "OMPI" for user-visible messages.  Thanks!

Reply:
Sure. Thanks!